spec/models/manageiq/providers/nuage/vsd_client/vcr_specs/vsd_client_spec.rb

```diff
@@ -0,0 +1,109 @@
+require 'json'
+describe ManageIQ::Providers::Nuage::NetworkManager::VsdClient do
+  before(:each) do
+    @userid = Rails.application.secrets.nuage_network.try(:[], 'userid') || 'NUAGE_USER_ID'
+    @password = Rails.application.secrets.nuage_network.try(:[], 'password') || 'NUAGE_PASSWORD'
+    @hostname = Rails.application.secrets.nuage_network.try(:[], 'host') || 'nuagenetworkhost'
+
+    # Ensure that VCR will obfuscate the basic auth
+    VCR.configure do |c|
+      # workaround for escaping host
+      c.before_playback do |interaction|
+        interaction.filter!(CGI.escape(@hostname), @hostname)
+        interaction.filter!(CGI.escape('NUAGE_NETWORK_HOST'), 'nuagenetworkhost')
+      end
+      c.filter_sensitive_data('NUAGE_NETWORK_AUTHORIZATION') { Base64.encode64("#{@userid}:#{@password}").chomp }
+    end
+  end
+
+  it "should return valid non empty json responses" do
+    VCR.use_cassette('manageiq/providers/nuage/network_manager/vsd_client/vsd_client', :allow_unused_http_interactions => true, :record => :new_episodes) do
+      vsd_client = ManageIQ::Providers::Nuage::NetworkManager::VsdClient.new("https://#{@hostname}:8443/nuage/api/v5_0", @userid, @password)
+
+      enterprises_resp = vsd_client.get_enterprises
+      domains_resp = vsd_client.get_domains
+      zones_resp = vsd_client.get_zones
+      subnets_resp = vsd_client.get_subnets
+      vports_resp = vsd_client.get_vports
+      vms_resp = vsd_client.get_vms
+      policy_groups_resp = vsd_client.get_policy_groups
+
```

Comment:
`end` the VCR.use_cassette block here just after you perform the API calls. The `expect` statements should be outside of that block.